pkg/pluginManager/controllerReconciler.go

```diff
@@ -43,26 +52,89 @@ func (s *pluginControllerReconciler) Reconcile(ctx context.Context, req reconcil
 		logger := s.logger.With(zap.String(instance.Kind, instance.Name))
 		logger.Sugar().Debugf("reconcile handle %v", instance)
 
+		// since we set ownerRef for the corresponding resource, we don't need to care about the cleanup procession
 		if instance.DeletionTimestamp != nil {
 			s.logger.Sugar().Debugf("ignore deleting task %v", req)
 			return ctrl.Result{}, nil
 		}
 
+		// create resource
+		{
+			var resource crd.TaskResource
+			var err error
+			var deletionTime *metav1.Time
+
+			if instance.Status.Resource == nil {
+				newScheduler := scheduler.NewScheduler(s.client, s.apiReader, KindNameNetReach, instance.Name, s.runtimeUniqueMatchLabelKey, logger)
+				resource, err = newScheduler.CreateTaskRuntimeIfNotExist(ctx, &instance, instance.Spec.AgentSpec)
+				if nil != err {
+					logger.Error(err.Error())
+					return ctrl.Result{}, err
+				}
+				instance.Status.Resource = &resource
+				logger.Sugar().Infof("try to update %s/%s status with resource %v", KindNameNetReach, instance.Name, resource)
+				err = s.client.Status().Update(ctx, &instance)
+				if nil != err {
+					logger.Error(err.Error())
+					return ctrl.Result{}, err
+				}
+			} else {
+				// avoid controller restarted
+				if instance.Status.FinishTime != nil {
+					deletionTime = instance.Status.FinishTime.DeepCopy()
+					if instance.Spec.AgentSpec.TerminationGracePeriodMinutes != nil {
+						newTime := metav1.NewTime(deletionTime.Add(time.Duration(*instance.Spec.AgentSpec.TerminationGracePeriodMinutes) * time.Minute))
+						deletionTime = newTime.DeepCopy()
+					}
+					logger.Sugar().Debugf("task finish time '%s' and runtime deletion time '%s'", instance.Status.FinishTime, deletionTime)
+				}
+			}
+
+			// let the tracker trace status
```

Comment:
// comment: the tracer will be asynchronous to update the resource status in the CR